src/appleseed-max-impl/appleseedblendmtl/appleseedblendmtl.cpp

```diff
@@ -0,0 +1,744 @@
+{
+    return 0.0f;
+}
+
+void AppleseedBlendMtl::SetAmbient(Color c, TimeValue t)
+{
+}
+
+void AppleseedBlendMtl::SetDiffuse(Color c, TimeValue t)
+{
+}
+
+void AppleseedBlendMtl::SetSpecular(Color c, TimeValue t)
+{
+}
+
+void AppleseedBlendMtl::SetShininess(float v, TimeValue t)
+{
+}
+
+void AppleseedBlendMtl::Shade(ShadeContext& sc)
+{
+}
+
+int AppleseedBlendMtl::NumSubMtls()
+{
+    return MtlCount; // MtlCount;
+}
+
+Mtl* AppleseedBlendMtl::GetSubMtl(int i)
+{
+    Mtl* mtl = nullptr;
+    if (i==0)
+        m_pblock->GetValue(ParamIdBaseMtl, 0, mtl, FOREVER);
+    else
+        m_pblock->GetValue(ParamIdLayerMtl, 0, mtl, FOREVER, i-1);
+    return mtl;
+}
+
+void AppleseedBlendMtl::SetSubMtl(int i, Mtl* m)
+{
+    if (i == 0)
+        m_pblock->SetValue(ParamIdBaseMtl, 0, m);
+    else
+        m_pblock->SetValue(ParamIdLayerMtl, 0, m, i-1);
+}
+
+MSTR AppleseedBlendMtl::GetSubMtlSlotName(int i)
+{
+    const auto mtl_id = static_cast<MtlId>(i);
+    return g_material_slot_names[mtl_id];
+}
+
+int AppleseedBlendMtl::get_sides() const
+{
+    return asr::ObjectInstance::BothSides;
+}
+
+bool AppleseedBlendMtl::can_emit_light() const
+{
+    return false;
+}
+
+asf::auto_release_ptr<asr::Material> AppleseedBlendMtl::create_material(
+    asr::Assembly&  assembly,
+    const char*     name,
+    const bool      use_max_procedural_maps)
+{
+    return
+        use_max_procedural_maps
+            ? create_builtin_material(assembly, name)
+            : create_osl_material(assembly, name);
+}
+
+asf::auto_release_ptr<asr::Material> AppleseedBlendMtl::create_osl_material(
+    asr::Assembly&  assembly,
+    const char*     name)
+{
+    Mtl* mat = nullptr;
+    m_pblock->GetValue(ParamIdBaseMtl, 0, mat, FOREVER);
+
+    if (!mat)
+        return asr::OSLMaterialFactory().create(name, asr::ParamArray());
+
+    auto shader_group_name = make_unique_name(assembly.shader_groups(), std::string(name) + "_shader_group");
+    auto shader_group = asr::ShaderGroupFactory::create(shader_group_name.c_str());
+
+    connect_sub_mtl(assembly, shader_group.ref(), name, "BaseMtl", mat);
+
+    TimeValue time = GetCOREInterface()->GetTime();
+    
+    int mat_count = m_pblock->Count(ParamIdLayerMtl);
+
+    asr::ParamArray shader_params;
+    int j = 0;
+    for (int i = 0; i < mat_count; ++i)
+    {
+        Mtl* mat = nullptr;
+        m_pblock->GetValue(ParamIdLayerMtl, time, mat, FOREVER, i);
+
+        if (!mat)
+            continue;
+
+        connect_sub_mtl(assembly, shader_group.ref(), name, asf::format("LayerMtl_{0}", j).c_str(), mat);
+
+        Texmap* tex = nullptr;
+        m_pblock->GetValue(ParamIdMaskTex, time, tex, FOREVER, i);
+        if (tex)
+        {
+            float mask_amount = m_pblock->GetFloat(ParamIdMaskAmount, time, FOREVER, i) / 100.0f;
+            connect_float_texture(shader_group.ref(), name, asf::format("MaskColor_{0}", j).c_str(), tex, mask_amount);
+        }
+
+        shader_params.insert(
```

Comment:
Let's simplify this a bit, for instance by moving the `GetColor()` call out of the expression.